packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -434,6 +435,39 @@ export default class ScriptTransformer {
     return fileSource;
   }
 
+  requireAndTranspileModule(
+    moduleContext: NodeModule,
+    moduleName: string,
+  ): any {
+    this.preloadTransformer(moduleName);
+
+    let transforming = false;
+    const revertHook = addHook(
+      (code, filename) => {
+        try {
+          transforming = true;
+          return this.transformSource(filename, code, false).code || code;
+        } finally {
+          transforming = false;
+        }
+      },
+      {
+        exts: [path.extname(moduleName)],
+        ignoreNodeModules: false,
+        matcher: (...args) => {
+          if (transforming) {
+            // Don't transform any dependency required by the transformer itself
+            return false;
+          }
+          return this.shouldTransform(...args);
+        },
+      },
+    );
+    const module = moduleContext.require(moduleName);
```

Comment:
Why do we use `moduleContext` here?

Reply:
`moduleName` would be relative to the file it was called from. I'm resolving it from the module it was called rather than inside `jest-transform`.

This seems to be not working inside the change in `jest-runtime`, hence the breaking test.

Reply:
that shouldn't be needed - we get an absolute path from config, so using plain `require` should work fine

Reply:
Thank you, it was an absolute nightmare trying debug why this wasn't working in `jest-runtime`. Work fine now